### DISCUSSION TITLE
Good Bluetooth dongle

### DISCUSSION THREAD

**Discussion by Arman-GeDon** [custom-components/ble_monitor#249] [Q&A]
Hi 
What good Bluetooth dongle do you recommend for use with that works in HA (in VirtualBox)? Especially with external antenna.
I've found this one on Ali but don't know if it's working with HA
https://www.aliexpress.com/item/4001079955049.html

**Comment by Magalex2x14** [comment]
Hello, @Arman-GeDon! Sorry for the delayed answer.
I use exactly this dongle with our HA integration without any problems (purchased at the end of 2020 from the supplier you specified).
Debian 10.8, Linux kernel 4.19

**Reply by Arman-GeDon** [reply]
Ok, good to know. Thank you

**Reply by mjvelten93** [reply]
Hi, could you recommend other Bluetooth Dongle for a single HA Intel Nuc (Gen 5) Integration (via USB). Struggling, which one to order (in the product description its always: no Linux). Thanks a lot!

**Comment by Magalex2x14** [comment]
@mjvelten93 as I already wrote, the dongle from the link above works without any problems for me on Linux (CSR8510 A10 chip). I think any with this chip will work.
I also have an ASUS USB-BT400 on a Broadcom chip - it also works, but it has a built-in antenna, not a full-size one.
I also remember mentioning TPLink UB400, also a built-in antenna, it seems to be a CSR8510 A10 chip (I can’t confirm).

The general recommendation is to try to find out what chip is used in the adapter and google its compatibility with Linux (or read customer reviews). Manufacturers often don't list Linux as a compatible OS because they mean official support only. I also think that the vast majority of BT4.0 adapters will work. You need to be careful with BT5.0 adapters, because their chipsets are newer, and their support in Linux may be raw or even absent (a lot depends on the kernel used in your linux distribution).

Preference should be given to adapters with an external full-size antenna (you can even install the bigger antenna).
<img width="200" alt="image" src="https://user-images.githubusercontent.com/4982025/154275353-4dc31795-8acf-4a43-b571-84a725ff6ca0.png">
